sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -2496,7 +2500,9 @@ class SQLConf extends Serializable with Logging {
   def legacyMsSqlServerNumericMappingEnabled: Boolean =
     getConf(LEGACY_MSSQLSERVER_NUMERIC_MAPPING_ENABLED)
 
-  def legacyTimeParserEnabled: Boolean = getConf(SQLConf.LEGACY_TIME_PARSER_ENABLED)
+  def legacyTimeParserEnabled: Boolean =
```

Comment:
This is used to create `DateFormatter` and `TimestampFormatter`, and I'm pretty sure these 2 formatters are used in many expressions and other places like the json parser.

I think it's better to put the logic in the formatter, not in some expressions.

Reply:
Thanks, got it. I moved this logic into formatter in cc9fd4f.